File: python/programmers/level 1/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    numlist = ['0','1','2','3','4','5','6','7','8','9']
    strlist = ['zero','one','two','three','four','five','six','seven','eight','nine']
    while not s.isdigit():
        for index in range(len(numlist)):
            s = s.replace(strlist[index], numlist[index])
            if s.isdigit():
                return int(s)
        
    return int(s)

File: java/src/main/java/ps_traning/programmers/level_1/숫자문자열과_영단어.java
package ps_traning.programmers.level_1;

public class 숫자문자열과_영단어 {
    public int solution(String s) {
        int answer = 0;
        String[] strArr = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        String[] numArr = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};

        for (int i = 0; i <= 9; i++) {
            s = s.replaceAll(strArr[i], numArr[i]);
        }
        answer = Integer.valueOf(s);
        return answer;
    }
}
